db/schema.sql

```diff
@@ -48,41 +48,71 @@ CREATE TABLE runencounter (
 DROP TABLE IF EXISTS hardware_ref;
 CREATE TABLE hardware_ref (
   id INT NOT NULL AUTO_INCREMENT,
-
+  item_name VARCHAR(255) NOT NULL,
+  icon VARCHAR(255) NOT NULL,
+  item_desc VARCHAR(255) NOT NULL,
+  mouseover VARCHAR(255) NOT NULL, -- ???
+  damage_type INT NOT NULL, -- or varchar?
+  strength INT NOT NULL,
+  rarity INT NOT NULL, -- or varchar?
   PRIMARY KEY(id)
 );
 
 DROP TABLE IF EXISTS software_ref;
 CREATE TABLE software_ref (
   id INT NOT NULL AUTO_INCREMENT,
-
+  card_name VARCHAR(255) NOT NULL,
+  icon VARCHAR(255) NOT NULL,
+  card_desc VARCHAR(255) NOT NULL,
+  damage_type INT NOT NULL, -- or varchar?
+  strength INT NOT NULL,
+  upgrade_level INT NOT NULL,
+  cooldown INT NOT NULL,
+  initial_cooldown INT NOT NULL,
+  rarity INT NOT NULL, -- or varchar?
   PRIMARY KEY(id)
 );
 
 DROP TABLE IF EXISTS encounter_ref;
 CREATE TABLE encounter_ref (
   id INT NOT NULL AUTO_INCREMENT,
-
+  act INT NOT NULL,
+  enemy1id_fk INT NOT NULL,
+  enemy2id_fk INT NOT NULL,
+  enemy3id_fk INT NOT NULL,
+  enemy4id_fk INT NOT NULL,
+  loot_rating INT NOT NULL,
   PRIMARY KEY(id)
 );
 
 DROP TABLE IF EXISTS enemy_ref;
 CREATE TABLE enemy_ref (
   id INT NOT NULL AUTO_INCREMENT,
-
+  enemy_name VARCHAR(255) NOT NULL,
+  enemy_desc VARCHAR(255) NOT NULL,
+  defence INT NOT NULL,
+  damage_type INT NOT NULL, -- or varchar?
+  health INT NOT NULL,
   PRIMARY KEY(id)
 );
 
 DROP TABLE IF EXISTS enemy_attack_ref;
 CREATE TABLE enemy_attack_ref (
   id INT NOT NULL AUTO_INCREMENT,
-
+  attack_name VARCHAR(255) NOT NULL,
+  attack_desc VARCHAR(255) NOT NULL,
+  effect INT NOT NULL, -- or varchar?
+  damage_type INT NOT NULL, -- or varchar?
+  strength INT NOT NULL,
+  turn_priority INT NOT NULL,
   PRIMARY KEY(id)
 );
 
 DROP TABLE IF EXISTS enemy_attack_list;
 CREATE TABLE enemy_attack_list (
   id INT NOT NULL AUTO_INCREMENT,
-
+  enemyid_fk INT NOT NULL,
+  attackid_fk INT NOT NULL,
   PRIMARY KEY(id)
 );
+-- is enemy attack list really needed if you just put enemyid_fk in enemy_attack_ref?
```

Comment:
The thinking is: we might want to have multiple enemies use the same attack, so that's more of a future-proofing than anything else.